nautobot/docs/models/extras/dynamicgroup.md

```diff
@@ -1,7 +1,319 @@
+
+Similarly, by the time we get to `third-child`, the filter at that node in the graph will have already had the filter from `first-child` and `second-child` applied, and will merely be passed through to the filter generated from the children of `third-child`.
+
+Because `first-child` and `second-child` are "included" by way of the `union` set operator (Boolean `OR`), but because the filter for `third-child` is empty as explained above, its filter condition is not directly included, resulting in `(first-child OR second-child)` as the first group, representing all _direct_ filters for the group `parent`.
+
+!!! note
+    But what about the `AND` coming from the association of `first-child`, you might be asking?
+
+    Well, that `AND` is also passed through because the default behavior when performing queryset filtering is to join all filter statements together with `AND`. For example, consider when you perform this from the Django ORM `Device.objects.filter(site__slug="ams01")` the outcome is in fact equivalent to `AND site__slug="ams01"`. Therefore, for the first child group (literally `first-child` in this case), initial `AND` will be omitted.
+
+Continuing on to the children of `third-child`, the same iteration rules apply. The filter from `nested-child` gets applied to the filter being passed through from its parent, `third-child`, except that the "exclude" (boolean `NOT`) operator is still applied from the association of `third-child` to parent resulting in `(NOT nested-child`).
+
+!!! note
+    You'll see `NOT` emitted as an `AND NOT` because `NOT` is actually just shorthand for this. They are in fact interchangeable as Boolean operators. While it's technically possible to perform an `OR NOT` query from a database perspective, it is not supported by Dynamic Groups because the `NOT` operator maps directly to the "difference" set operation which is distinctly an `AND NOT` Boolean operation when performing underlying SQL queries at the database.
+
+Piecing the rules together, we ended up with two filters and by wrapping them in their own set of parentheses we get our final generated filter:
+
+```no-highlight
+((first-child OR second-child) AND (NOT nested-child))
+```
+
+How does this work?
+
+| Start with   | Filter        | Operator | Weight | Group Count | Final Count |
+| ------------ | ------------- | -------- | ------ | ----------- | ----------- |
+| parent       | none          | none     | none   | 389         | 389         |
+| first-child  | site=ams01    | restrict | 10     | 11          | 11          |
+| second-child | site=ang01    | include  | 20     | 7           | 18          |
+| third-child  | none          | none     | 30     | 389         | n/a         |
+| nested-child | status=active | exclude  | 10     | 389         | 0           |
```

Comment:
- This table is using restrict/include/exclude, while the previous table is using intersection/union/difference - would be good to be consistent.
- Group Count and Final Count aren't discussed anywhere above and so suddenly introducing them here confuses things rather than clarifying them, I think

Reply:
This is a WIP. I have no idea where to put this.